plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -194,6 +221,61 @@ public IcebergTableHandle getTableHandle(ConnectorSession session, SchemaTableNa
                 Optional.ofNullable(nameMappingJson));
     }
 
+    @Override
+    public boolean isSupportedVersionType(ConnectorSession session, SchemaTableName tableName, PointerType pointerType, io.trino.spi.type.Type versioning)
+    {
+        switch (pointerType) {
+            case TEMPORAL:
+                return versioning instanceof TimestampWithTimeZoneType || versioning instanceof TimestampType;
+            case TARGET_ID:
+                return versioning == BIGINT;
+        }
+        throw new TrinoException(NOT_SUPPORTED, "Version type " + versioning + " is not supported in Iceberg");
+    }
+
+    private Optional<Long> getSnapshotIdFromVersion(Table table, Optional<ConnectorTableVersion> versionOptional)
```

Comment:
Return doesn't have to be optional, all cases either return Optional.of(), or throw

Reply:
updated